GVFS/GVFS.Common/Git/GitSslSettings.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace GVFS.Common.Git
+{
+    public struct GitSslSettings
+    {
+        public readonly string SslCertificate;
+        public readonly bool SslCertPasswordProtected;
+
+        public GitSslSettings(IDictionary<string, GitConfigSetting> configSettings)
+        {
+            if (configSettings != null)
+            {
+                if (configSettings.TryGetValue(GitConfigSetting.SslCert, out var sslCerts))
+                {
+                    this.SslCertificate = sslCerts.Values.Single();
+                }
+                else
+                {
+                    this.SslCertificate = default(string);
+                }
+
+                if (configSettings.TryGetValue(GitConfigSetting.SslCertPasswordProtected, out var isSslCertPasswordProtected))
+                {
+                    this.SslCertPasswordProtected = isSslCertPasswordProtected.Values.Select(bool.Parse).Single();
+                }
+                else
+                {
+                    this.SslCertPasswordProtected = false;
+                }
+            }
+            else
+            {
+                this.SslCertificate = default(string);
+                this.SslCertPasswordProtected = default(bool);

```

Comment:
Let's be explicit about `null` and `false` as the default values.

Reply:
Great suggestion, changes made.